### PR TITLE
#15 angular 15

### LINKED ISSUE #15: Update to angular@15

### PR DESCRIPTION
**Pull request by simonmumenthaler**
closes #15

### CODE REVIEW

**Comment by simonmumenthaler** [comment]
Libs `@angular-builders/jest` and `jest-preset-angular` for angular@^15 are not yet released.

**Comment by michaelwittwer** [comment]
@simonmumenthaler ng15 support for `@angular-builders/jest` was beta release and `jest-preset-angular` is stable, updated the deps in commits.

**Comment by simonmumenthaler** [comment]
current beta/pr versions:
- `@angular-builders/jest`  (`15.0.0-beta.1`)
- `@shiftcode/utilities` / `@shiftcode/branch-utilities` / `@shiftcode/publish-helper`  (`2.0.0-pr14.5`)

**Comment by simonmumenthaler** [comment]
Logger now supports the provide/feature patter for standalone bootstrapping:

image

File: libs/ssr/src/lib/determine-origin.function.ts [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
+      throw new Error(`env var ${envVarName} was not set or is empty`)
     }
-    const protocol = isAwsLambdaEnv() ? 'https' : request.protocol
-    return `${protocol}://${domain}`
+    return `https://${domain}`
   } else {
-    throw new Error('can not determinate the origin - make sure the module is only applied for SSR?')
+    const request = inject(REQUEST)
+    // we have to add the port to the hostname if we're running local
+    return `${request.protocol}://${request.hostname}:4000` // FIXME: use the actual port
```
Comment (michaelwittwer) [inline libs/ssr/src/lib/determine-origin.function.ts:24]: any ideas how to fix the FIXME?

File: tsconfig.json [hunk 2]
```diff
@@ -34,14 +34,15 @@
     "experimentalDecorators": true,
     "moduleResolution": "node",
     "importHelpers": true,
-    "target": "es2020",
+    "target": "ES2022",
     "module": "es2020",
     "lib": [
       "es2020",
```
Comment (michaelwittwer) [inline tsconfig.json:40]: should also be ES2022

**Comment by michaelwittwer** [comment]
@simonmumenthaler please update READMEs for aws & ssr libs to reflect the new style of providing stuff

**Review by michaelwittwer — APPROVED** [review]

**Pull request merged**